diff -u -r ../cache/go/src/runtime/runtime1.go ./src/runtime/runtime1.go
--- ../cache/go/src/runtime/runtime1.go	2026-02-10 01:22:00.000000000 +0000
+++ ./src/runtime/runtime1.go	2026-03-13 22:52:38.612222594 +0000
@@ -5,6 +5,7 @@
 package runtime
 
 import (
+	"cgocfg"
 	"internal/bytealg"
 	"internal/goarch"
 	"internal/runtime/atomic"
@@ -57,6 +58,16 @@
 	argv **byte
 )
 
+func AndroidSelfExecutable() string {
+	var adjust int32
+	if cgocfg.Enabled {
+		adjust = 0
+	} else {
+		adjust = 1
+	}
+	return gostringnocopy(argv_index(argv, adjust))
+}
+
 // nosplit for use in linux startup sysargs.
 //
 //go:nosplit
@@ -74,9 +79,19 @@
 	if GOOS == "windows" {
 		return
 	}
-	argslice = make([]string, argc)
-	for i := int32(0); i < argc; i++ {
-		argslice[i] = gostringnocopy(argv_index(argv, i))
+	var adjust int32
+	if GOOS == "android" && !cgocfg.Enabled {
+		adjust = 1
+	} else {
+		adjust = 0
+	}
+	argslice = make([]string, argc - adjust)
+	for i := int32(0); i < argc - adjust; i++ {
+		idx := i
+		if i > 0 {
+			idx += adjust
+		}
+		argslice[i] = gostringnocopy(argv_index(argv, idx))
 	}
 }
 
